### PR TITLE
fix wrong framework retry records 

### LINKED ISSUE #4911: Task retry will cause wrong framework retry records
How to reproduce:

1. Use this job config:

```yaml
protocolVersion: 2
name: retry_test_3_a9a85a47
type: job
jobRetryCount: 10
prerequisites:
  - type: dockerimage
    uri: 'openpai/standard:python_3.6-pytorch_1.2.0-gpu'
    name: docker_image_0
taskRoles:
  taskrole:
    instances: 2
    completion:
      minFailedInstances: 1
      minSucceededInstances: -1
    taskRetryCount: 2
    dockerImage: docker_image_0
    resourcePerInstance:
      gpu: 1
      cpu: 1
      memoryMB: 2000
    commands:
      - printenv
      - exit 133
defaults:
  virtualCluster: default
extras:
  gangAllocation: false
```

2. Go to the job retry page:

image

Some tasks are recorded in `RUNNING` status. It is not true.

The cause for this problem:

FC will output framework object every time a task is retried. But the fluentd only records the framework object when the first task retry happens.

### PR DESCRIPTION
**Pull request by hzy46**
to fix #4911

### CODE REVIEW

**Comment by hzy46** [comment]
Test result (use the same config in #4911)

image

File: src/fluentd/deploy/fluentd-config.yaml.template [hunk 1]
```diff
@@ -94,6 +94,24 @@ data:
...
     </filter>
 
+    # add logType
+    <filter fc-objectsnapshots>
+      @type record_transformer
+      enable_ruby
+      <record>
+        logType "${\
+          if record['logMessage'].include? 'enqueueFrameworkObj: Framework Pod Deleted'
+            'deleted_pod_snapshot'
```
Comment (yqwang-ms) [inline src/fluentd/deploy/fluentd-config.yaml.template:104]: >deleted_pod_snapshot [](start = 13, length = 20)

pod_deleted

File: src/fluentd/deploy/fluentd-config.yaml.template [hunk 2]
```diff
@@ -94,6 +94,24 @@ data:
...
+          elsif record['logMessage'].include? 'syncFrameworkState: Framework will be retried'
+            'retried_framework_snapshot'
```
Comment (yqwang-ms) [inline src/fluentd/deploy/fluentd-config.yaml.template:106]: >retried_framework_snapshot [](start = 13, length = 26)

framework_retried

File: src/fluentd/deploy/fluentd-config.yaml.template [hunk 3]
```diff
@@ -94,6 +94,24 @@ data:
...
+          elsif record['logMessage'].include? 'syncTaskState: Task will be retried'
+            'task_retried_framework_snapshot'
```
Comment (yqwang-ms) [inline src/fluentd/deploy/fluentd-config.yaml.template:108]: >task_retried_framework_snapshot [](start = 13, length = 31)

task_retried

File: src/fluentd/deploy/fluentd-config.yaml.template [hunk 4]
```diff
@@ -94,6 +94,24 @@ data:
       </record>
     </filter>
 
+    # add logType
+    <filter fc-objectsnapshots>
+      @type record_transformer
+      enable_ruby
+      <record>
+        logType "${\
+          if record['logMessage'].include? 'enqueueFrameworkObj: Framework Pod Deleted'
```
Comment (yqwang-ms) [inline src/fluentd/deploy/fluentd-config.yaml.template:103]: >'enqueueFrameworkObj: Framework Pod Deleted [](start = 43, length = 43)

'Framework Pod Deleted'

File: src/fluentd/deploy/fluentd-config.yaml.template [hunk 5]
```diff
@@ -94,6 +94,24 @@ data:
...
+            'deleted_pod_snapshot'
+          elsif record['logMessage'].include? 'syncFrameworkState: Framework will be retried'
```
Comment (yqwang-ms) [inline src/fluentd/deploy/fluentd-config.yaml.template:105]: >Framework will be retried' [](start = 67, length = 26)

'Framework will be retried'

File: src/fluentd/deploy/fluentd-config.yaml.template [hunk 6]
```diff
@@ -94,6 +94,24 @@ data:
...
+            'retried_framework_snapshot'
+          elsif record['logMessage'].include? 'syncTaskState: Task will be retried'
```
Comment (yqwang-ms) [inline src/fluentd/deploy/fluentd-config.yaml.template:107]: >Task will be retried' [](start = 62, length = 21)

'Task will be retried'

File: src/fluentd/deploy/fluentd-config.yaml.template [hunk 7]
```diff
@@ -94,6 +94,24 @@ data:
       </record>
     </filter>
 
+    # add logType
```
Comment (yqwang-ms) [inline src/fluentd/deploy/fluentd-config.yaml.template:97]: >logType [](start = 10, length = 7)

objectSnapshotTrigger

**Review by yqwang-ms — APPROVED** [review]
:shipit:

**Pull request merged**